src/sink_http.c

```diff
@@ -216,7 +216,7 @@ static int serialize_metrics(struct http_sink* sink, metrics* m, void* data) {
 
     int post_len = 0;
     char* post_data = strbuf_get(post_buf, &post_len);
-    strbuf_free(post_buf, false);
+    strbuf_free(post_buf, true);
```

Comment:
The leak is likely later in the code, somewhere after lifoq dequeue and in the curl error handling path